### DISCUSSION TITLE
Custom comparison for specific elements in XML

### DISCUSSION THREAD

**Discussion by mgnslndh** [VerifyTests/Verify#1022] [Q&A]
Is it possible to compare XML snapshots with custom comparison for specific elements (perhaps via XPath)? For example, if a value in a node is allowed to diff ±5.

**Comment by SimonCropp** [comment]
can you do a select via XPath, then compare the result?

**Reply by mgnslndh** [reply]
It would be possible to write specific test cases. The current solution is using a custom implementation to compare the XML files but its not that good and the context is kind of lost when there is a diff. I thought it might be a good fit for snapshot testing use VerifyTests but there is this rule where some values are allowed to diff slightly. 

I was looking at the implementation of VerifyXml and it looks like eventually it is down to a string comparison, which is reasonable for snapshot testing. But that also means that the element/attribute context is lost and I do not see a way of plugging in a custom comparison. Perhaps this is too much of an edge case. 

A dirty trick could be to pre-process all "actual" value to the "expected" value if they are in the allowed diff range. And then do the snapshot test. So, actually similar to the scrubbing process.
